src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -243,6 +264,36 @@ private BitsOrPostingList bitsOrPostingListForKeyRange(QueryContext context, Abs
         }
     }
 
+    private int[] findApproximateMatches(QueryContext context, int limit, long minSSTableRowId, long maxSSTableRowId, PrimaryKeyMap primaryKeyMap, OrdinalsView ordinalsView, NodeSimilarity.ApproximateScoreFunction scoreFunction) throws IOException
+    {
+        PriorityQueue<Pair<Integer, Float>> pq = new PriorityQueue<>(limit + 1, Comparator.comparing(Pair::right));
```

Comment:
PQ has a pretty high overhead compared to just dropping everything in an array and sorting, even though it's "less work" quicksort is just too good

So I'm not 100% sure that this is slower than a simple sort, I'm maybe 60% sure

P.S. if sorting is in fact faster, should update ArrayPostingList to take a length parameter so we can return the top K w/o having to copy into a smaller array

Reply:
what's the typical number of rows in sstable? 
PQ limits memory we spent + it has predictable max.
Otherwise I'll have to allocate array of (maxSSTableRowId-minSSTableRowId) or let arraylist resize.
PQ keeps memory in check if limit is significantly smaller than maxSSTableRowId-minSSTableRowId.